test/fmf/tests/fdo-bootc.sh

```diff
@@ -0,0 +1,260 @@
+#!/bin/bash
+set -euox pipefail
+
+# FDO test in RHEL-10 Image Mode
+
+# Local variables
+VM_NAME="fdo-disk-image"
+GUEST_ADDRESS=192.168.100.50
+SSH_USER="admin"
+
+# Set up temporary files.
+TEMPDIR=$(mktemp -d)
+
+# SSH setup.
+SSH_OPTIONS=(-o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o ConnectTimeout=5)
+SSH_KEY=key/ostree_key
+EDGE_USER_PASSWORD=foobar
+
+function clean_up {
+    greenprint "🧼 Cleaning up"
+    sudo rm -rf ./{Containerfile,output,config.toml}
+    sudo podman logout registry.stage.redhat.io
+    sudo podman rmi -a -f
+    sudo rm -rf /var/lib/libvirt/images/{fdo-iso.qcow2,install.iso}
+    sudo virsh net-destroy integration && sudo virsh net-undefine integration
+    VM_STATUS=$(sudo virsh list --all | awk '{print $3}' | tail -n 2)
+    if [[ $VM_STATUS == running ]]; then
+        sudo virsh destroy "${VM_NAME}"
+    fi
+    sudo virsh undefine "${VM_NAME}" --nvram
+    # TO DO: Remove tag from quay.io registry
+    # skopeo delete --creds "${QUAY_USERNAME}:${QUAY_PASSWORD}" "${QUAY_REPO_URL}:${QUAY_REPO_TAG}"
+}
+
+# Colorful output.
+function greenprint {
+    echo -e "\033[1;32m${1}\033[0m"
+}
+
+wait_for_ssh_up () {
+    SSH_STATUS=$(sudo ssh "${SSH_OPTIONS[@]}" -i "${SSH_KEY}" "${SSH_USER}@${1}" '/bin/bash -c "echo -n READY"')
+    if [[ $SSH_STATUS == READY ]]; then
+        echo 1
+    else
+        echo 0
+    fi
+}
+
+check_result () {
+    greenprint "🎏 Checking for test result"
+    if [[ $RESULTS == 1 ]]; then
+        greenprint "💚 Success"
+    else
+        greenprint "❌ Failed"
+        clean_up
+        exit 1
+    fi
+}
+
+trap clean_up EXIT
+
+sudo dnf install -y \
+    cargo \
+    openssl \
+    openssl-devel \
+    git \
+    make \
+    systemd \
+    krb5-devel \
+    python3-docutils \
+    gpgme-devel \
+    libassuan-devel \
+    systemd-rpm-macros \
+    rpmdevtools \
+    golang \
+    go-rpm-macros \
+    python3-devel \
+    selinux-policy-devel \
+    device-mapper-devel \
+    podman \
+    qemu-img \
+    qemu-kvm \
+    libvirt-client \
+    libvirt-daemon-kvm \
+    libvirt-daemon \
+    virt-install
+
+# Clone upstream repo and build fdo packages
+echo "Build fdo packages"
+cd /tmp/"${REPO_NAME}"
+make rpm
+sudo dnf -y install rpmbuild/RPMS/x86_64/*
+cd -
+mkdir -pv packages
+sudo cp /tmp/"${REPO_NAME}"/rpmbuild/RPMS/x86_64/* ./packages
+
+# Manufacturing server setup
+DOWNLOAD_NODE="${DOWNLOAD_NODE}" ./setup-manufacturing-server.sh
```

Comment:
This script is configuring all the FDO services, not only the manufacturing server. I would rename it to something more meaningful `setup-fdo-servers.sh` o similar.

Reply:
Script's name has been modified as suggested.